File: tests/output/jpa/src/main/javagen/restaurant/jpa_identity_enums/dtos/restaurant/RestaurantItem.java
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

package restaurant.jpa_identity_enums.dtos.restaurant;

import jakarta.annotation.Generated;

/**
 * Détail d'un restaurant en liste.
 */
@Generated("TopModel : https://github.com/klee-contrib/topmodel")
public interface RestaurantItem {

	/**
	 * Identifiant du restaurant.
	 */
	Integer getId();

	/**
	 * Nom du restaurant.
	 */
	String getNom();

	/**
	 * Adresse du restaurant.
	 */
	String getAdresse();

	/**
	 * Numéro de téléphone.
	 */
	String getTelephone();

	/**
	 * Hydrate values of instance.
	 * @param id value to set.
	 * @param nom value to set.
	 * @param adresse value to set.
	 * @param telephone value to set.
	 */
	void hydrate(Integer id, String nom, String adresse, String telephone);
}

File: tests/output/vanilla-js/model/restaurant/restaurant-item.ts
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

import {DO_ID, DO_LIBELLE, DO_TELEPHONE} from "../../domains";

export interface RestaurantItem {
    id: number;
    nom: string;
    adresse?: string;
    telephone?: string;
}

export const RestaurantItemEntity = {
    id: {
        type: "field",
        name: "id",
        domain: DO_ID,
        isRequired: true,
        label: "Id"
    },
    nom: {
        type: "field",
        name: "nom",
        domain: DO_LIBELLE,
        isRequired: true,
        label: "Nom"
    },
    adresse: {
        type: "field",
        name: "adresse",
        domain: DO_LIBELLE,
        isRequired: false,
        label: "Adresse"
    },
    telephone: {
        type: "field",
        name: "telephone",
        domain: DO_TELEPHONE,
        isRequired: false,
        label: "Telephone"
    }
} as const;
